src/server/system_services/cluster_server.js

```diff
@@ -135,7 +135,7 @@ function pre_add_member_to_cluster(req) {
                 dbg.log0('updating adding server ip in db');
                 let shard_idx = cutil.find_shard_index(req.rpc_params.shard);
                 let server_idx = _.findIndex(topology.shards[shard_idx].servers,
-                    server => server.address === '127.0.0.1');
+                    server => server.address === 'localhost');
```

Comment:
I think we need a `net_utils.is_localhost(address)` to check both 127.0.0.1 or localhost or ::1
because in these cases where we check we probably want to accept all options.